site/app/controllers/AuthenticationController.php

```diff
@@ -308,4 +310,191 @@ public function userSelection() {
 
         return new WebResponse(AuthenticationView::class, 'userSelection', $users);
     }
+
+    public function sendVerificationEmail(string $email, string $verification_code, string $user_id): void {
+        $subject = "Submitty Email Verification";
+        $url = $this->core->getConfig()->getBaseUrl() . 'authentication/verify_email?verification_code=' . $verification_code;
+        $body = <<<EMAIL
+Welcome to Submitty! We are excited to have you on board. To complete your account setup, either enter this verification code, or click the link below.
+
+Verification Code: $verification_code
+
+Verification Link: $url
+
+If you didn't sign up for Submitty, you can ignore this email.
+
+Welcome,
+Submitty Team
+EMAIL;
+
+        $details = [
+            'subject' => $subject,
+            'body' => $body,
+            'email_address' => $email,
+            'to_name' => $user_id
+        ];
+        $email = new Email($this->core, $details);
+        $this->core->getNotificationFactory()->sendEmails([$email]);
+    }
+
+    /**
+     * Display the form for creating a new account
+     */
+    #[Route("/authentication/create_account", methods: ['GET'])]
+    public function signupForm(): ResponseInterface {
+        // Check if the user is already logged in, if yes, redirect to home or another appropriate page
+        if ($this->logged_in) {
+            return new RedirectResponse($this->core->buildUrl(['home']));
+        }
+        if (!$this->core->getConfig()->isUserCreateAccount()) {
+            $this->core->addErrorMessage('Users cannot create their own account, Please have your system administrator add you.');
+            return new RedirectResponse($this->core->buildUrl(['authentication', 'login']));
+        }
+        return new WebResponse('Authentication', 'signupForm', ['email' => $this->core->getConfig()->getAcceptedEmails(), 'user_id' => $this->core->getConfig()->getUserIdRequirements()]);
```

Comment:
This line is pretty long.  Wrap it instead?

Also, `email` and `user_id` seem a bit misleading here.